src/cuckoo.c

```diff
@@ -220,89 +253,93 @@ static uint8_t *Filter_FindUnique(CuckooBucket bucket, size_t index, uint16_t bu
         *err = CuckooInsert_NoSpace;
     }
     return firstEmpty;
-}
+}*/
 
-static CuckooInsertStatus Filter_KOInsert(CuckooBucket *curFilter, size_t numBuckets,
-                                          uint16_t maxIterations, uint16_t bucketSize,
-                                          const LookupParams *params, LookupParams *victim);
+static CuckooInsertStatus Filter_KOInsert(CuckooFilter *filter, SubCF *curFilter, 
+                                          const LookupParams *params);
 
 static CuckooInsertStatus CuckooFilter_InsertFP(CuckooFilter *filter, const LookupParams *params) {
-    CuckooBucket *curFilter = filter->filters[filter->numFilters - 1];
-    uint8_t *slot = Filter_FindAvailable(curFilter, filter->bucketSize, params);
-    if (slot) {
-        *slot = params->fp;
-        filter->numItems++;
-        return CuckooInsert_Inserted;
+    for(size_t ii = filter->numFilters; ii > 0; --ii) {
```

Comment:
Is this strictly needed for this change? I guess if we 'promote' to a larger filter, then that filter out to get used more, rather than starting from the first filter..

Reply:
@mnunberg 
The thinking here was, a filled slot in a small filter contributes more to the error rate than a filled slot in a large filter. Therefore, if we open a larger filter, we should attempt to use it first.